### ISSUE TITLE
Setup after view is unloaded and reloaded is not performed

### ISSUE THREAD

**Issue by tonyarnold** [iridia/IRSlidingSplitViewController#3]
I just hit a bug in one of my projects where the root view controller's view (an instance of `IRSlidingSplitViewController`) would be removed from super due to memory constraints, but not setup properly again when it was setup again.

Basically, the master and detail view controllers are not being re-added to the view hierarchy, so I did the following:

``` objective-c

- (void)viewDidLoad
{
    [super viewDidLoad];

    BOOL showingMasterViewController = self.showingMasterViewController;
    UIViewController *mVC = self.masterViewController;
    UIViewController *dVC = self.detailViewController;

    self.masterViewController = nil;
    self.detailViewController = nil;

    self.masterViewController = mVC;
    self.detailViewController = dVC;
}
```

It's ugly, but it works — a more elegant solution will be forthcoming when I have some free time.

**Comment by evadne** [comment]
OMG.

That is unacceptable.

Fixing.

**Comment by evadne** [comment]
Thanks for your contribution.

This issue has been identified and fixed on `develop` at 15114c9.  One way to prove that it is fixed, is to force view unloading by doing this in your own subclass:

```
- (void) didReceiveMemoryWarning {

    [super didReceiveMemoryWarning];

    if ([self isViewLoaded])
    if (!self.view.window)
        self.view = nil;

}
```

Then, present a view controller modally over the sliding split view controller and trigger a simulated memory warning.  You should be able to verify that without the fix, the view is always blanked upon re-appearing, but with the fix in place, the view is not blanked upon re-appearing.

The fix also corrects an issue where the replaced view controllers will inadvertently have their nonexistent view loaded just for removal.

Another fix in efffaa6 corrects a belated issue where when the sliding split view’s view is offscreen, it still triggers implicit view creation just for housekeeping.  As long as the sliding split view controller is either used as a root view controller or properly contained using view controller containment, it should work.

I’ll be closing the issue as of now, but feel free to re-open it in the future if anything exploded.

**Comment by tonyarnold** [comment]
Nice, thanks @monoceroi!
